Ghidra should ignore macOS dot-underscore files in Application.findFilesByExtensionInApplication()

ghidra issue

**Describe the bug**
This issue comes up when I open Ghidra.

When I open Ghidra on my macOSX, it explodes with tonnes of error dialog boxes.Most of them are of the ``` Validation error``` form followed by some ``` ... languages/ [filename].ldefs ```. Each dialog box states the same error message except the [filename]. Some example of these filenames : 

._AARCH64.ldefs
._8085.ldefs
._8051.ldefs
._8048.ldfes

and many more. The error dialog box for [filename]=._z80.ldefs is 

``` 
Validation error: Invalid language definitions file: /Volumes/BACKUP/SANDISK DATA/ghidra_9.1-BETA_DEV/Ghidra/Processors/Z80/data/languages/._z80.ldefs ```

Full description of the bug is here : 
Validation error: Invalid language definitions file: /Volumes/BACKUP/SANDISK DATA/ghidra_9.1-BETA_DEV/Ghidra/Processors/Z80/data/languages/._z80.ldefs
ghidra.app.plugin.processors.sleigh.SleighException: Invalid language definitions file: /Volumes/BACKUP/SANDISK DATA/ghidra_9.1-BETA_DEV/Ghidra/Processors/Z80/data/languages/._z80.ldefs
	at ghidra.app.plugin.processors.sleigh.SleighLanguageValidator.validateSleighFile(SleighLanguageValidator.java:108)
	at ghidra.app.plugin.processors.sleigh.SleighLanguageValidator.validateLdefsFile(SleighLanguageValidator.java:73)
	at ghidra.app.plugin.processors.sleigh.SleighLanguageProvider.createLanguages(SleighLanguageProvider.java:82)
	at ghidra.app.plugin.processors.sleigh.SleighLanguageProvider.createLanguages(SleighLanguageProvider.java:76)
	at ghidra.app.plugin.processors.sleigh.SleighLanguageProvider.<init>(SleighLanguageProvider.java:66)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at ghidra.util.classfinder.ClassSearcher.getInstances(ClassSearcher.java:140)
	at ghidra.util.classfinder.ClassSearcher.getInstances(ClassSearcher.java:126)
	at ghidra.program.util.DefaultLanguageService.searchForProviders(DefaultLanguageService.java:89)
	at ghidra.program.util.DefaultLanguageService.<init>(DefaultLanguageService.java:79)
	at ghidra.program.util.DefaultLanguageService.getLanguageService(DefaultLanguageService.java:51)
	at ghidra.program.database.ProgramDB.<init>(ProgramDB.java:300)
	at ghidra.program.database.ProgramContentHandler.getDomainObject(ProgramContentHandler.java:190)
	at ghidra.framework.data.GhidraFileData.getDomainObject(GhidraFileData.java:338)
	at ghidra.framework.data.GhidraFile.getDomainObject(GhidraFile.java:179)
	at ghidra.app.util.task.OpenProgramTask.openFileMaybeUgrade(OpenProgramTask.java:241)
	at ghidra.app.util.task.OpenProgramTask.openUnversionedFile(OpenProgramTask.java:206)
	at ghidra.app.util.task.OpenProgramTask.openDomainFile(OpenProgramTask.java:149)
	at ghidra.app.util.task.OpenProgramTask.run(OpenProgramTask.java:133)
	at ghidra.util.task.Task.monitoredRun(Task.java:126)
	at ghidra.util.task.TaskRunner.lambda$startTaskThread$1(TaskRunner.java:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:178)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1471)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:978)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
	at org.iso_relax.verifier.impl.VerifierImpl.verify(VerifierImpl.java:121)
	at ghidra.app.plugin.processors.sleigh.SleighLanguageValidator.validateSleighFile(SleighLanguageValidator.java:103)
	... 26 more

---------------------------------------------------
Build Date: 2019-Sep-23 1119 EDT
Ghidra Version: 9.1-BETA
Java Home: /Volumes/BACKUP/SANDISK DATA/jdk/jdk-11.0.6.jdk/Contents/Home
JVM Version: Oracle Corporation 11.0.6
OS: Mac OS X 10.15.4 x86_64
Workstation: jugnu-da-mac-2.hitronhub.home
```

**To Reproduce and Behavior**
Some Background information. I have placed all the Ghidra files on my SSD(save space on my local disk) and placed a symbolic link on my Desktop. 

1. I click the ghidraRun symbolic link to actual ghidraRun in my SSD.
2. A shell spawns and it open Ghidra main screen (Dragon), opening Project dialog box and then FileBrowser.
3. I select a 32 bit ELF executable and it spawns CodeBrowser followed by many Validation error dialog boxes.
4. I hold Enter and it presses Ok on current and all subsequent dialog boxes.
5. As explained in above, each dialog box is Validation Error about some language file. 
6. Everything is as usually 

I guess because I do not use many features, so I do not know If I am missing anything because of these boxes. I only do automatical analysis to produced a decompiled C like output of the binary. 


**Screenshots**
I have made some screenshots with prefix numi_ .. to follow up with the chronology of opening the Ghidra. 

** NOTE : Attachments**
I do not have knowledge of Ghidra internals, so if there are any backlogs I can check, please let me know and I will update this issue

<img width="1278" alt="num7_press_enter_and_get_rid_of_boxes" src="https://user-images.githubusercontent.com/22536182/79200254-01d8e680-7deb-11ea-9a9a-3d239931b8a7.png">


**Environment (please complete the following information):**
OS information
```
Darwin jugnu-da-mac-2.hitronhub.home 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```
Ghidra information
Version 9.1-BETA

JAVA : 

```
➜  ~ /Library/Internet\ Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java -version
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b11, mixed mode)
```

**Additional context**
Add any other context about the problem here.
<img width="1255" alt="num1_ghidra_files_on_my_ssd" src="https://user-images.githubusercontent.com/22536182/79200240-fe455f80-7dea-11ea-9bb3-2f5ca42460df.png">
<img width="1025" alt="num2_symbolic_link_to_ghidraRun" src="https://user-images.githubusercontent.com/22536182/79200242-ff768c80-7dea-11ea-8664-4218149c1885.png">
<img width="796" alt="num3_fileSelectorBox" src="https://user-images.githubusercontent.com/22536182/79200245-ff768c80-7dea-11ea-9acd-dd1d8406ddb0.png">
<img width="1280" alt="num4_multiple_dialog_box_pops_up" src="https://user-images.githubusercontent.com/22536182/79200247-000f2300-7deb-11ea-89b3-ecefff83d41f.png">
<img width="1271" alt="num5_dialogBoxExample" src="https://user-images.githubusercontent.com/22536182/79200248-00a7b980-7deb-11ea-8d28-dfaa008fd2e7.png">
<img width="1278" alt="num6_many_dialog_boxes" src="https://user-images.githubusercontent.com/22536182/79200250-00a7b980-7deb-11ea-8216-d04aaf01e6ae.png">

--- a/Ghidra/Framework/Utility/src/main/java/ghidra/framework/GModule.java
+++ b/Ghidra/Framework/Utility/src/main/java/ghidra/framework/GModule.java
@@ -161,6 +161,12 @@ public class GModule {
 				}
 			}
 			else {
+
+				// Ignore ._ resource fork files
+				if (child.getName().startsWith("._")) {
+					continue;
+				}
+
 				if (child.getName().endsWith(extension)) {
 					accumulator.add(child);
 				}

